core-build/gulp-core-build/src/index.ts

```diff
@@ -45,16 +45,16 @@ export * from './tasks/CleanTask';
 export * from './tasks/ValidateShrinkwrapTask';
 export * from './jsonUtilities/SchemaValidator';
 
-// tslint:disable-next-line:no-any
-const packageJSON: any = require(path.resolve(process.cwd(), 'package.json'));
 const _taskMap: { [key: string]: IExecutable } = {};
 const _uniqueTasks: IExecutable[] = [];
 
-const packageFolder: string =
-  (packageJSON.directories && packageJSON.directories.packagePath) ?
-    packageJSON.directories.packagePath : '';
+const packageFolder: string = (builtPackage.directories && builtPackage.directories.packagePath)
+  ? builtPackage.directories.packagePath
+  : '';
 
 let _buildConfig: IBuildConfig = {
+  gulp: undefined as any, // tslint:disable-line:no-any
```

Comment:
`buildConfig.gulp` is non-optional because tasks should expect it to be defined, but the actual instance of `gulp` is passed in through the `gcb.initialize` function below. So here we do want it to be undefined (because it'll be defined later), but we don't want the type to be nullable. Same for `rootPath`.

I'll add a comment explaining this.

---
In reply to: [124910073](https://github.com/Microsoft/web-build-tools/pull/256#discussion_r124910073) [](ancestors = 124910073,124909572)